packages/api/lambdas/reports/orca-backup-reconciliation-report.js

```diff
@@ -235,40 +377,54 @@ async function orcaReconciliationReportForGranules(recReportParams) {
   };
 
   const collectionsConfig = await fetchCollectionsConfig(recReportParams);
-  log.debug(`fetchESCollections returned ${JSON.stringify(collectionsConfig)}`);
-
-  const esSearchParams = convertToESGranuleSearchParamsWithCreatedAtRange(recReportParams);
-  log.debug(`Create ES granule iterator with ${JSON.stringify(esSearchParams)}`);
-  const esGranulesIterator = new ESSearchQueue(
-    {
-      ...esSearchParams,
-      sort_key: ['granuleId', 'collectionId'],
-    },
-    'granule',
-    process.env.ES_INDEX
+  log.debug(`fetchCollections returned ${JSON.stringify(collectionsConfig)}`);
+
+  const knex = await getKnexClient();
+  const searchParams = convertToDBGranuleSearchParams(recReportParams);
+
+  const granulesSearchQuery = getGranulesByApiPropertiesQuery({
+    knex,
+    searchParams,
+    sortByFields: ['granule_id', 'collectionName', 'collectionVersion'],
+    temporalBoundByCreatedAt: true,
+  });
+
+  log.debug(`Create PG granule iterator with ${granulesSearchQuery}`);
+
+  const pgGranulesIterator = new QuerySearchClient(
+    granulesSearchQuery,
+    100 // arbitrary limit on how items are fetched at once
```

Comment:
Reviewers: looking for opinions on value/approach here.   Given we're explicitly bounding this on performance testing per team decision/given existing limitations in the tool, I'm not sure if that veers into YAGNI given this might need refactored.